api_client.py

```diff
@@ -533,34 +531,43 @@ def get_partner_organization_tenants(self, whoami_response, access_token):
         Returns:
             dict -- response containing whoami response or error
         """
-        tenants = {}
-        try:
-            default_headers = {
-                "Authorization": "Bearer " + access_token,
-                "X-Partner-ID": whoami_response["id"],
-            }
-            tenant_url = (
-                whoami_response["apiHosts"]["global"]
-                + "/"
-                + whoami_response["idType"]
-                + "/v1/tenants"
-            )
-            tenant_response = self.request_url(tenant_url, None, default_headers, 1)
+        if self.config.tenant_id:
+            tenant = {}
+            try:
+                if whoami_response["idType"] == "organization":
+                    default_headers = {
+                        "Authorization": "Bearer " + access_token,
+                        "X-Organization-ID": whoami_response["id"],
+                    }
+                else:
+                    default_headers = {
+                        "Authorization": "Bearer " + access_token,
+                        "X-Partner-ID": whoami_response["id"],
+                    }
+                    
+                page_no = 1
+                tenant_url = (
+                    whoami_response["apiHosts"]["global"]
+                    + "/"
+                    + whoami_response["idType"]
+                    + "/v1/tenants/"
+                    + self.config.tenant_id
+                )
+                tenant_response = self.request_url(tenant_url, None, default_headers, 1)
 
-            self.log("Tenant response: %s" % tenant_response)
-            tenants = json.loads(tenant_response)
-            tenants["items"] = list(
-                filter(lambda tenant: tenant["id"] == self.config.tenant_id, tenants["items"])
-            )
-            if len(tenants["items"]) > 0:
-                return tenants
-            else:
+                self.log("Tenant response: %s" % (tenant_response))
+                tenant = json.loads(tenant_response)
+                return tenant
+                    
+            except json.decoder.JSONDecodeError as e:
+                self.log("Sophos partner tenant API response not in json format")
+                return {"error": e}
+            except Exception as e:
+                self.log("Error :: %s" % e)
                 raise Exception(
-                    "Configuration file mention tenant id not matched with whoami data tenant id"
+                     f"Configured Tenant ID is not managed by {whoami_response['idType']} whose credentials are provided"
```

Comment:
I'm not sure you can assert that it is an access issue in response to a generic exception.

Reply:
Can you please suggest the text for this?

Reply:
It needs to be something generic like: `f"Error getting tenant {tenant_id}"` with the exception object giving more information.

Reply:
Updated the message with a display exception object